src/cli/shared/CicsDeployProfile.option.ts

```diff
@@ -0,0 +1,28 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright IBM Corp, 2019
+*
+*/
+
+import { ICommandOptionDefinition } from "@brightside/imperative";
+
+/**
+ * Imperative option for the "cics-deploy-profile" parameter
+ *
+ */
+export const CicsDeployProfileOption: ICommandOptionDefinition = {
```

Comment:
I don't think you need to define this option yourself, I think Imperative does it for you. See for example https://github.com/zowe/zowe-cli-cics-plugin/blob/master/src/cli/define/program/Program.definition.ts

It also mentions here that the --xxx-profile option is autogenerated
https://github.com/zowe/imperative/wiki/Working-With-User-Profiles

Reply:
Correct... but it's nice to have it listed as a normal parameter in the help text.

Reply:
Looks like the cics-plugin does duplicate most of the Imperative config for the options in two different places. It's inline within the Imperative.ts file, but also defined within src/cli/CicsSession.ts. There are subtle differences between the two, for example the profile has -p as an alias for password, but the command line has --pw instead (I guess -p might clash with the alias for port in the full command line). Some of the options are defined in CicsSession and then get referenced from Imperative.ts, that referencing mechanism might work for us... I'll give it a go.

Reply:
Turns out that single source is possible, but you can't have a context sensitive description that way. If you have two separate copies then you can say, for example, "--cicsplex specifies the CICSplex to target" in the profile, but "--cicsplex specifies the CICSplex to target if you have not set a profile" in the command line help. And so forth. I guess that's why the cics plugin duplicates much of the config, it's so that subtle differences can be deliberately introduced between the two usage patterns.